awx/main/tasks.py

```diff
@@ -147,6 +147,8 @@ def handle_setting_changes(self, setting_keys):
         if key.startswith('LOG_AGGREGATOR_'):
             restart_local_services(['uwsgi', 'celery', 'beat', 'callback'])
             break
+        elif key == 'OAUTH2_PROVIDER':
+            restart_local_services(['uwsgi'])
```

Comment:
I feel like it would be possible to change these settings without a restart -- maybe a custom subclass of `OAuth2ProviderSettings`? Not likely to change often though, so maybe not a big deal.